lib/services/role.go

```diff
@@ -2079,6 +2102,33 @@ func (set RoleSet) EnhancedRecordingSet() map[string]bool {
 	return m
 }
 
+// HostUsers returns host user information matching a server or nil if
+// a role disallows host user creation
+func (set RoleSet) HostUsers(s types.Server) *HostUsersInfo {
+	var groups []string
+	serverLabels := s.GetAllLabels()
+	for _, role := range set {
+		result, _, err := MatchLabels(role.GetNodeLabels(types.Allow), serverLabels)
```

Comment:
I think you probably need to check the deny labels as well, the role may not actually match this node. You may be able to make use of `CheckAccess` actually by making a RoleSet with the single role

Reply:
Yes, we need to check against deny rules also. See `CheckKubeGroupsAndUsers` for an example.

Reply:
@lxea I believe this still needs to be addressed

Reply:
sorry, I didn't see a comment or a commit message, didn't notice that you added that, looks good now